### ISSUE TITLE
[data grid] Keyboard navigation on header cell fails if focused cell is not rendered

### ISSUE THREAD

**Issue by romgrk** [mui/mui-x#19359]
### Search keywords

focus column header keybord navigation

### Latest version

- [x] I have tested the latest version

### Steps to reproduce

Focus a header cell, the use the keyboard arrows to navigate.

<img width="1047" height="254" alt="Image" src="https://github.com/user-attachments/assets/1060efe8-f90c-447e-b59f-c9471957bd0d" />

Noticed while testing #19323, but the issue seems old (present even in v7).

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>
